### PR TITLE
Identation and refactoring

### PR DESCRIPTION
**Pull request by OolonColoophid**
Hi Tomas

Some changes as requested. I created separated branches for each of these locally, then merged them into my master before uploading.

I've tweaked (all of these are in my commit messages):

- Improved indentation throughout
- Separated functions where necessary
- Renamed some variables

Cheers

Ian

### CODE REVIEW

File: bot.js [hunk 1]
```diff
@@ -64,6 +64,15 @@ function matchPattern(p, w, pi, wi) {
 function saySomething() {
 	var message = document.getElementById('message').value;
 	var conversation = document.getElementById('conversation');
+
+	answer = reply(message);
```
Comment (tpetricek) [inline bot.js:71]: It would probably be better to add `var` here:

```
var answer = reply(message);
```

(But this is not really related to this class, just generally to JavaScript!)

File: bot.js [hunk 2]
```diff
@@ -82,7 +85,7 @@ function reply(message) {
 		var match = matchPattern(rules[i].pattern, words, 0, 0)
 		if (match) {
 			// Reconstruct a reply - if a token is a number, find the
-			// matched word from 'match', otherwise just append the word
+			// matched word from 'match', otherwordIndexse just append the word
```
Comment (tpetricek) [inline bot.js:88]: I guess this is search & replace error :-) This is why good editors have a refactoring tool that actually understand your code and don't replace just strings!

**Comment by tpetricek** [comment]
Nice work! I added some minor comments, just FYI.

File: bot.js (hunk 1 above)
Comment (OolonColoophid) [reply bot.js:71]: Oh yes - thanks, Tomas

**Comment by OolonColoophid** [comment]
Thanks, Tomas - more haste less speed!

**Comment by OolonColoophid** [comment]
Great - thanks for looking at this.
